WcaOnRails/app/controllers/api/v0/competitions_controller.rb

```diff
@@ -1,11 +1,55 @@
 # frozen_string_literal: true
 class Api::V0::CompetitionsController < Api::V0::ApiController
-  def show
-    competition = Competition.where(id: params[:id], showAtAll: true).first
-    unless competition
-      render json: { error: "Competition with id #{params[:id]} not found" }, status: 404
-      return
+  def index
+    managed_by_user = nil
+    if params[:managed_by_me].present?
+      require_scope!("manage_competitions")
+      managed_by_user = current_api_user
     end
+
+    competitions = Competition.search(params[:q], params: params, managed_by_user: managed_by_user)
```

Comment:
I was just thinking this simple inlining. 

```
  def index
    if params[:managed_by_me].present?
      require_scope!("manage_competitions")
    end

    competitions = Competition.search(params[:q], params: params, managed_by_user: current_api_user)
    competitions = competitions.includes(:delegates, :organizers)

    paginate json: competitions
  end
```
If `current_api_user` can return something even when `params[:managed_by_me].blank?`, I see now it's incorrect.